### PR TITLE
Android tests new package

### PR DESCRIPTION
**Pull request by ranjith-zen3**
# Checklist
- [x] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-java/blob/master/.github/CONTRIBUTING.md).
- [x] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [x] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [x] This pull-request is submitted against the `master` branch. 

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

# Description of the solution

### CODE REVIEW

File: iot-e2e-tests/android/app/src/androidTest/java/com/iothub/azure/microsoft/com/android/SendMessagesIT.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
+/*
+ *  Copyright (c) Microsoft. All rights reserved.
+ *  Licensed under the MIT license. See LICENSE file in the project root for full license information.
+ */
+
+package com.iothub.azure.microsoft.com.android;
```
Comment (prmathur-microsoft) [inline iot-e2e-tests/android/app/src/androidTest/java/com/iothub/azure/microsoft/com/android/SendMessagesIT.java]: Rename appropriately to match others com.microsoft.azure.sdk.iot.android

File: iot-e2e-tests/android/app/src/main/java/com/iothub/azure/microsoft/com/android/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ */
+
+package com.iothub.azure.microsoft.com.android;
+
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+
+public class MainActivity extends AppCompatActivity {
+
+
```
Comment (prmathur-microsoft) [inline iot-e2e-tests/android/app/src/main/java/com/iothub/azure/microsoft/com/android/MainActivity.java]: remove extra ;lines

File: iot-e2e-tests/android/app/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:paddingBottom="@dimen/activity_vertical_margin"
+    android:paddingLeft="@dimen/activity_horizontal_margin"
+    android:paddingRight="@dimen/activity_horizontal_margin"
+    android:paddingTop="@dimen/activity_vertical_margin"
+    tools:context="com.iothub.azure.microsoft.com.android.com.iothub.azure.microsoft.com.android.MainActivity">
+
+</RelativeLayout>
```
Comment (prmathur-microsoft) [inline iot-e2e-tests/android/app/src/main/res/layout/activity_main.xml]: are the below .png needed for checkin ?

File: iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/iothubservices/SendMessagesCommon.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+				if (!messageSent.getResult()) {
+					Assert.fail("Sending message over AMQPS protocol failed");
+				}
+			} catch (Exception e) {
+				Assert.fail("Sending message over AMQPS protocol failed");
+			}
+		}
+
+		client.closeNow();
```
Comment (prmathur-microsoft) [inline iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/iothubservices/SendMessagesCommon.java]: remove extra lines

File: iot-e2e-tests/jvm/pom.xml [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+                        <configuration>
+                            <shadedArtifactAttached>true</shadedArtifactAttached>
+                            <shadedClassifierName>with-deps</shadedClassifierName>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+</project>
```
Comment (prmathur-microsoft) [inline iot-e2e-tests/jvm/pom.xml]: please make sure all the files from current iot-e2e-tests folder is moved to this. I see only partial movement to jvm

File: iot-e2e-tests/android/app/src/androidTest/java/com/iothub/azure/microsoft/com/android/SendMessagesIT.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    // How much to wait until a message makes it to the server, in milliseconds
+    private static final Integer SEND_TIMEOUT_MILLISECONDS = 60000;
+
+    //How many milliseconds between retry
+    private static final Integer RETRY_MILLISECONDS = 100;
+
+    //Device Connection String
+    private String connString = "";
+
+
```
Comment (prmathur-microsoft) [inline iot-e2e-tests/android/app/src/androidTest/java/com/iothub/azure/microsoft/com/android/SendMessagesIT.java]: Can you please add setup and tear down and all of those details in android tests too. It should use same environmental variables

File: iot-e2e-tests/android/app/src/main/res/layout/activity_main.xml (hunk 3 above)
Comment (ranjith-zen3) [reply iot-e2e-tests/android/app/src/main/res/layout/activity_main.xml]: yes those are required

File: iot-e2e-tests/jvm/pom.xml (hunk 5 above)
Comment (ranjith-zen3) [reply iot-e2e-tests/jvm/pom.xml]: all the files are moved

File: service/iot-service-client/pom.xml [hunk 7]
```diff
@@ -3,7 +3,7 @@
     <groupId>com.microsoft.azure.sdk.iot</groupId>
     <artifactId>iot-service-client</artifactId>
     <name>Iot Hub Java Service SDK</name>
-    <version>1.8.24</version>
+    <version>1.8.25</version>
     <description>The Microsoft Azure IoT Service SDK for Java</description>
```
Comment (prmathur-microsoft) [inline service/iot-service-client/pom.xml]: Why was this change needed ?

File: iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/SendMessagesIT.java [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+    private static final Integer SEND_TIMEOUT_MILLISECONDS = 60000;
+
+    //How many milliseconds between retry
+    private static final Integer RETRY_MILLISECONDS = 100;
+
+    //Device Connection String
+    private String connString = "";
+
+
+    @Test
```
Comment (prmathur-microsoft) [inline iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/SendMessagesIT.java]: Remove extra lines 40

File: iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/SendMessagesIT.java [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+
+    // How much to wait until a message makes it to the server, in milliseconds
+    private static final Integer SEND_TIMEOUT_MILLISECONDS = 60000;
+
+    //How many milliseconds between retry
+    private static final Integer RETRY_MILLISECONDS = 100;
+
+    //Device Connection String
+    private String connString = "";
+
```
Comment (prmathur-microsoft) [inline iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/SendMessagesIT.java]: I understand that service client does not work currently, but how are you fetching environmental variables for these Android tests. I don't see any code for that.

Comment (prmathur-microsoft) [reply iot-e2e-tests/android/app/src/androidTest/java/com/microsoft/azure/sdk/iot/android/SendMessagesIT.java]: Please add relevant code in setup and teardown excluding the use of service client if that is an issue.

**Comment by prmathur-microsoft** [comment]
Please sign CLA for me to accept this PR. 
Also please squash all your commits into a sigle one and rebase to latest master as there are conflicts. Gate will not accept it. There are some extra e2e tests added just today. Please include them as well. 
Also our latest service client version is 1.9.24. Please include that see if it resolves your issue.

File: iot-e2e-tests/pom.xml [hunk 10]
```diff
@@ -1,116 +1,34 @@
...
     <modelVersion>4.0.0</modelVersion>
     <groupId>com.microsoft.azure.sdk.iot</groupId>
-    <artifactId>iot-e2e-tests</artifactId>
-    <name>IoT Hub Java SDK tests</name>
-    <version>0.10.13</version>
-    <description>Test suite fot the Microsoft Azure IoT Device SDK for Java</description>
+	<artifactId>iot-e2e-tests</artifactId>
+	<name>IoT Hub Java SDK tests</name>
+	<version>0.9.13</version>
+	<description>Test suite fot the Microsoft Azure IoT Device SDK for Java</description>
```
Comment (prmathur-microsoft) [inline iot-e2e-tests/pom.xml]: Please make sure all the versions correctly reflect what is present on master branch

File: iot-e2e-tests/pom.xml [hunk 11]
```diff
@@ -1,116 +1,34 @@
...
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-compiler-plugin</artifactId>
                 <version>3.3</version>
                 <configuration>
-                    <source>1.7</source>
-                    <target>1.7</target>
+                    <source>1.8</source>
+                    <target>1.8</target>
                 </configuration>
```
Comment (prmathur-microsoft) [inline iot-e2e-tests/pom.xml]: Why was this change needed ? would 1.7 not run when it was running earlier ?

File: iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/EventCallback.java [hunk 12]
```diff
@@ -22,8 +20,7 @@ public void execute(IotHubStatusCode status, Object context)
         if (status.equals(expectedStatusCode))
         {
             success.setResult(true);
-        }
-        else
+        } else
```
Comment (prmathur-microsoft) [inline iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/EventCallback.java]: revert

File: iot-e2e-tests/pom.xml [hunk 13]
```diff
@@ -1,56 +1,23 @@
...
     <groupId>com.microsoft.azure.sdk.iot</groupId>
-    <artifactId>iot-e2e-tests</artifactId>
-    <name>IoT Hub Java SDK tests</name>
-    <version>0.10.13</version>
-    <description>Test suite fot the Microsoft Azure IoT Device SDK for Java</description>
+	<artifactId>iot-e2e-tests</artifactId>
+	<name>IoT Hub Java SDK tests</name>
+	<version>0.10.13</version>
+	<description>Test suite fot the Microsoft Azure IoT Device SDK for Java</description> 
+    <packaging>pom</packaging>
```
Comment (prmathur-microsoft) [inline iot-e2e-tests/pom.xml]: Why is packaging POM needed ?

**Comment by prmathur-microsoft** [comment]
Closing this PR as #147 was a duplicate of this.
